### PR TITLE
[CA-1429] Remove parameters for v2 only

### PR DESCRIPTION
**Pull request by s-rubenstein**
This removes `privateIpGoogleAccess` and `highSecurityNetwork` from the v2 create billing project API, since those are going to be on by default in the new PPW world.

- [ ] **Submitter**: Include the JIRA issue number in the PR description
- [ ] **Submitter**: Check that the **Product Owner** has signed off on any user-facing changes
- [ ] **Submitter**: Make sure Swagger is updated if API changes
  - [ ] **...and Orchestration's Swagger too!**
- [ ] **Submitter**: If updating admin endpoints, also update [firecloud-admin-cli](https://github.com/broadinstitute/firecloud-admin-cli)
- [ ] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [ ] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * To Demo flag is set
  * Release Summary is filled out, if applicable
  * Add notes on how to QA
- [ ] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [ ] **Submitter**: Database checks:
  * If PR includes new or changed db queries, include the explain plans in the description
  * Make sure liquibase is updated if appropriate
  * If doing a migration, take a backup of the
  [dev](https://console.cloud.google.com/sql/instances/terraform-qfarbdq3lrexxck5htofjs5z6m/backups?project=broad-dsde-dev&organizationId=548622027621)
  and
  [alpha](https://console.cloud.google.com/sql/instances/terraform-r4caezzc35c4tb7pgdhwkmme4y/backups?project=broad-dsde-alpha&organizationId=548622027621)
  DBs in Google Cloud Console
- [ ] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [ ] Tell your tech lead (TL) that the PR exists if they want to look at it
- [ ] Anoint a lead reviewer (LR). **Assign PR to LR**
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [ ] **TL** sign off
- [ ] **LR** sign off
- [ ] **Assign to submitter** to finalize
- [ ] **Submitter**: Verify all tests go green, including CI tests
- [ ] **Submitter**: Squash commits and merge to develop
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Verify swagger UI on dev environment still works after deployment
- [ ] **Submitter**: Inform other teams of any API changes via Slack and/or email
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

**Comment by s-rubenstein** [comment]
Jenkins retest

**Comment by s-rubenstein** [comment]
Jenkins retest

**Comment by andy7i** [comment]
could you add the ticket # and a description of changes? I think this ticket also included some open questions so the answers to those should also be documented somewhere

**Comment by s-rubenstein** [comment]
Ah, I missed the questions in the comments of the ticket. Anything that is critical like that should probably end up in the body of the ticket, in the future. I will check in with all those folks and document.

**Comment by andy7i** [comment]
thanks for adding in more details on the PR!

it's a bit hidden, but from the body of the [ticket](https://broadworkbench.atlassian.net/browse/CA-1429):
>If enableFlowLogs should be on by default, then we can remove all three and use a single RBS pool, containing projects with all 3 enabled. (I think this would also remove the need for a separate pool for service perimeter projects.)
>
>If we keep enableFlowLogs as a parameter, we will need to thread that parameter through so that we get a project from the correct pool. (Bonus: We may also want to rename the pool so that it's clear that it isn't used only for service perimeter projects.)

I think we still need to handle this second pool. The fact that tests pass without it is probably not a good sign.. Maybe worth pulling that into a separate ticket / PR?

**Comment by s-rubenstein** [comment]
Yeah, that feels like a separate thread of work. I can pull that out into another PR and keep it in this ticket? (That feels fine to me at least)

File: core/src/test/scala/org/broadinstitute/dsde/rawls/webservice/BillingApiServiceV2Spec.scala [hunk 1]
```diff
@@ -271,58 +264,6 @@ class BillingApiServiceV2Spec extends ApiServiceSpec with MockitoSugar {
...
-      sealRoute(services.billingRoutesV2) ~>
-      check {
-        assertResult(StatusCodes.Created) {
-          status
-        }
-      }
-  }
-
-  it should "return 201 when creating a project with a highSecurityNetwork with enableFlowLogs and privateIpGoogleAccess turned on" in withEmptyDatabaseAndApiServices { services =>
-    val projectName = RawlsBillingProjectName("test_good")
```
Comment (andy7i) [inline core/src/test/scala/org/broadinstitute/dsde/rawls/webservice/BillingApiServiceV2Spec.scala:314]: do we have tests covering projects with enableFlowLogs turned on? Or will you cover this in the subsequent PR?

**Review by andy7i — APPROVED** [review]
1 question about tests, but if test cases exist or will exist, then 👍

File: core/src/main/resources/swagger/api-docs.yaml [hunk 2]
```diff
@@ -5845,6 +5845,29 @@ components:
...
+        servicePerimeter:
+          type: string
+          description: The fully qualified name of the GCP service perimeter to put
+            this project into in the form accessPolicies/[POLICY NUMBER]/servicePerimeters/[NAME].
+            Caller must have the add_project action for this service perimeter in
+            Sam.
+        enableFlowLogs:
+          type: boolean
+          description: Optional, false if not specified. If true, enable flow logs
+            within the high security network. Requires highSecurityNetwork = true.
```
Comment (marctalbott) [inline core/src/main/resources/swagger/api-docs.yaml]: nit: no need to include `Requires highSecurityNetwork = true`

**Review by marctalbott — COMMENTED** [review]
We should remove these parameters from `develop` as well, not just the feature branch. The v2 billing API is already merged to `develop`, so this PR should be merged into `develop` not the feature branch.

In a separate but related PR to the feature branch, we should update the create workspace call to pull from different RBS pools depending on if we want flow logs enabled.

File: core/src/test/scala/org/broadinstitute/dsde/rawls/webservice/BillingApiServiceV2Spec.scala (hunk 1 above)
Comment (s-rubenstein) [reply core/src/test/scala/org/broadinstitute/dsde/rawls/webservice/BillingApiServiceV2Spec.scala:314]: We do have tests here: https://github.com/broadinstitute/rawls/pull/1483/files#diff-874ce0961fead919b34abf612968996e00aa7ebecfeed0ba7aa4e34c7f587814R238. They don't test any special behavior though, and maybe they should.

**Comment by s-rubenstein** [comment]
Jenkins retest

**Comment by s-rubenstein** [comment]
Jenkins retest

**Pull request closed without merging**